### ISSUE TITLE
Model config and training

### ISSUE THREAD

**Issue by chaowentao** [daniegr/EfficientPose#23]
I am interested in your work. And I want to implement your model training. But I don't find the related files. So can you upload files in detail? Thank you.

**Comment by daniegr** [comment]
@chaowentao: First of all, great to get to know your interest in this project! Unfortunately, we are not at the moment planning to launch training code for EfficientPose. However, all necessary details are available in our paper (https://link.springer.com/article/10.1007/s10489-020-01918-7), which enables reimplementation.

**Comment by chaowentao** [comment]
@daniegr What is the loss of the article? paf and three-stage heatmap loss?

**Comment by daniegr** [comment]
@chaowentao: For all three detection passes (pass 1 with Part Affinity Fields and pass 2-3 with confidence maps) Euclidean loss is used, with appropriate sigma values to define the ground truth maps. As mentioned in Appendix B of the paper, a stepwise sigma scheme is utilized.

**Comment by JozefColdenhoff** [comment]
Great work, I also had a few questions regarding reimplementation.

(1) In the paper you mention that you evaluate EfficientPose by using "MSE of the model predictions relative to the target coordinates". However you mention here that Euclidean loss was used, was MSE then used only for the comparison of EfficientPose and OpenPose?
(2) Im guessing that the learningrates employed in the trainingscheme mentioned in Appendix B are specifically designed for an MSE loss function. Is there a formula that would allow for the recalculation of the scheme based on the highest learning rate for which a Euclidean loss function did not diverge? 
(3) Were the EfficientNet feature extractors frozen during the training?

Thanks in advance.

**Comment by daniegr** [comment]
@chaowentao: Thank you for your interest in the EfficientPose framework. To clarify and to avoid possible confusion with the previous comment, the description in Appendix B of the paper is correct. The training loss is implemented as the mean squared error (MSE) of the predicted maps _Y_ (i.e., Part Affinity Fields or confidence maps) relative to the ground truth maps _Y*_ (as determined by the ground truth locations and the sigma value) across the output responses _N_: 

<p align="center"><img align="center" src="https://i.upmath.me/svg/MSE%20%3D%20%5Cfrac%7B1%7D%7BN%7D%5Csum_%7Bi%3D1%7D%5E%7BN%7D%7B(Y_%7Bi%7D%20-%20Y%5E%7B*%7D_%7Bi%7D)%5E2%7D" alt="MSE = \frac{1}{N}\sum_{i=1}^{N}{(Y_{i} - Y^{*}_{i})^2}" /></p> 

Regarding your final question, the feature extractors were also fine-tuned during training.
